modules/statistics/server/jobs/daily-statistics.server.job.js

```diff
@@ -65,6 +65,139 @@ module.exports = function (job, agendaDone) {
+          },
+          tags: {
+            access: 'members'
+          }
+        }, done)
+      });
+    },
+
+    // Get the amount of members who logged into TR today
+    function (done) {
+      var stat = 'today';
+      statistics.getLastSeenStatistic(stat, function (err, count) {
+        if (err) {
+          return done(err);
+        }
+        writeDailyStat({
+          namespace: stat,
+          values: {
+            count: parseInt(count)
+          },
+          tags: {
+            access: 'members'
+          }
+        }, done)
+      });
+    },
+
+    // Get the amount of members who logged into TR this week
+    function (done) {
+      var stat = 'week';
+      statistics.getLastSeenStatistic(stat, function (err, count) {
+        if (err) {
+          return done(err);
+        }
+        writeDailyStat({
+          namespace: stat,
+          values: {
+            count: parseInt(count)
+          },
+          tags: {
+            access: 'members'
+          }
+        }, done)
+      });
+    },
+
+    // Get the amount of member who logged in this month
+    function (done) {
+      var stat = 'thisMonth';
+      statistics.getLastSeenStatistic(stat, function (err, count) {
+        if (err) {
+          return done(err);
+        }
+        writeDailyStat({
+          namespace: stat,
+          values: {
+            count: parseInt(count)
+          },
+          tags: {
+            access: 'members'
+          }
+        }, done)
+      });
+    },
+
+    // Get the ammount of members who logged in for that last month
+    function (done) {
+      var stat = 'sinceAMonthAgo';
+      statistics.getLastSeenStatistic(stat, function (err, count) {
+        if (err) {
+          return done(err);
+        }
+        writeDailyStat({
+          namespace: stat,
+          values: {
+            count: parseInt(count)
+          },
+          tags: {
+            access: 'members'
+          }
+        }, done)
+      });
+    },
+
+    // Get the ammount of members who logged in for the last 6 months
+    function (done) {
+      var stat = 'sinceSixMonthsAgo';
+      statistics.getLastSeenStatistic(stat, function (err, count) {
+        if (err) {
+          return done(err);
+        }
+        writeDailyStat({
+          namespace: stat,
+          values: {
+            count: parseInt(count)
+          },
+          tags: {
+            access: 'members'
+          }
+        }, done)
+      });
+    },
+
+    // Get the ammount of members who logged in for the last 12 months
+    function (done) {
+      var stat = 'sinceAYearAgo';
+      statistics.getLastSeenStatistic(stat, function (err, count) {
+        if (err) {
+          return done(err);
+        }
+        writeDailyStat({
+          namespace: stat,
+          values: {
+            count: parseInt(count)
+          },
+          tags: {
+            access: 'members'
+          }
+        }, done)
+      });
+    },
+
```

Comment:
There's quite a bit of code-repetition here; would looping them differently work better while ensuring that each task is done before the next one starts so that we don't have lots of expensive queries and writes on the air simultaneously?

Reply:
Yes. There is a code repetition for evey time-stat we have.
If each task is async how could I be sure when to start the next one?

I mean. I could do a look that interates into an array of stats I want to collect. But in that case my poor knowlege of JS would get confused on "what about this return(done)?" What should I have in the case of a loop? I need some help here with the theory...

Reply:
It indeed gets a bit elaborate with ES5 Javascript and callbacks and how this whole file is structured, so maybe I'd just do this for now:

```
function (done) {
  collectLastSeen(7, done);
},
function (done) {
  collectLastSeen(14, done);
},
```

```js
function collectLastSeen(seenSinceDays, callback) {
  statistics.getLastSeenStatistic(seenSinceDays, function (err, count) {
    if (err) {
      return callback(err);
    }
    writeDailyStat({
      namespace: stat,
      values: {
        count: parseInt(count)
      },
      tags: {
        access: 'members'
      }
    }, callback);
  });
}
```

We can simplify this whole file later on when [we're on ES6 Javascript](https://github.com/Trustroots/trustroots/issues/804) and can get rid of our async helpers completely.